pyasic/web/epic.py

```diff
@@ -0,0 +1,118 @@
+#                                                                              -
+#      http://www.apache.org/licenses/LICENSE-2.0                              -
+#                                                                              -
+#  Unless required by applicable law or agreed to in writing, software         -
+#  distributed under the License is distributed on an "AS IS" BASIS,           -
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.    -
+#  See the License for the specific language governing permissions and         -
+#  limitations under the License.                                              -
+# ------------------------------------------------------------------------------
+import json
+import warnings
+from typing import Union
+
+import httpx
+
+from pyasic import settings
+from pyasic.web import BaseWebAPI
+
+
+class ePICWebAPI(BaseWebAPI):
+    def __init__(self, ip: str) -> None:
+        super().__init__(ip)
+        self.username = "root"
+        self.pwd = settings.get("default_epic_password", "letmein")
+        self.token = None
+
+    async def send_command(
+        self,
+        command: Union[str, bytes],
+        ignore_errors: bool = False,
+        allow_warning: bool = True,
+        **parameters: Union[str, int, bool],
+    ) -> dict:
+        async with httpx.AsyncClient() as client:
+            is_get = False
+            for i in range(settings.get("get_data_retries", 1)):
+                try:
+                    if parameters.get("post"):
+                        parameters.pop("post")
+                        epic_param = {"param": parameters.get("parameters"), "password": self.pwd}
+                        response = await client.post(
+                            f"http://{self.ip}:4028/{command}",
+                            timeout=5,
+                            json=epic_param,
+                        )
+                    elif not parameters == {}:
+                        response = await client.post(
+                            f"http://{self.ip}:4028/{command}",
+                            timeout=5,
+                            json=parameters,
+                        )
+                    else:
+                        is_get = True
+                        response = await client.get(
+                            f"http://{self.ip}:4028/{command}",
+                            timeout=5,
+                        )
+                    if not response.status_code == 200:
+                        continue
+                    json_data = response.json()
+                    if json_data:
+                        # The API can return a fail status if the miner cannot return the requested data. Catch this and pass
+                        if "result" in json_data and json_data["result"] is False and is_get:
+                            raise json.JSONDecodeError("The miner returned a fail status.", doc=json_data, pos=0)
```

Comment:
This should use `pyasic.APIError` with the result message, such as `raise APIError(json_data["msg"])` if that key exists.